javascript/ql/src/semmle/javascript/frameworks/ClientRequests.qll

```diff
@@ -230,7 +239,12 @@ private class SuperAgentUrlRequest extends CustomClientRequest {
   }
 
   override DataFlow::Node getADataNode() {
-    none()
+    exists (DataFlow::MethodCallNode chained, string name |
+      name = "set" or name = "send" or name = "query" |
+      chained.getReceiver*() = this and
```

Comment:
Perhaps we could introduce `SourceNode.getAMemberCall()` (implemented as `getAMemberCall(_)`) and then phrase this as `chained = getAMemberCall*().getAMethodCall(name)`? Alternatively, introducing a new predicate to model the chaining also seems justified.

Reply:
Done, but based on `getAMethodCall` instead of `getAMemberCall`.
I looked through the other mentions of `chained` in the code base for refactoring opportunities, but none of the cases were trivial to rewrite to use the new `getAChainedMethodCall`.